### PR TITLE
ci: update mergify rules to include teams

### LINKED ISSUE #2367: Improve contribution workflow to speed up issue tracking and PR merging
This proposal aims to improve the workflow for accepting contributions to the project.

## Current Bottlenecks

Maintainers of the project are the only ones that can add labels to issues and PRs, update project boards and manage the merging of PRs.

## Suggested Improvement

There is a group of _core-contributors_ that regularly

- handles reported Issues
- send PRs
- reviews PRs
- works on Issues and PRs associated with different project boards

It would be beneficial for the Ceph-CSI project that the _core-contributors_ get the privileges

- to assign labels to Issues and PRs
- have their PR reviews count
- update project boards

PRs should get merged when there is a approved review of a _core-contributor_ and at least one _maintainer_. This will improve the velocity of merging PRs, as currently two _maintainers_ need to approve PRs. The approvals of _core-contributors_ are not counted by Mergify and only used as input for the review of _maintainers_.

## Required Changes

By using GitHub teams, Mergify can be configured to require a review from a _core-contributor_ and a _maintainer_.

### GitHub Settings

There need to be two GitHub teams:

- a team `core-contributors` with `write` permissions
- a team `maintainers` with `write` permissions

A _maintainer_ should be part of both teams, as the _maintainer_ is still a _core-contributor_. It might be possible to make the `maintainers` team a sub-team of `core-contributors`.

### Mergify Rules

The current `#approved-reviews-by>=2` rule will allow two approvals from _core-contributors_ once the team has been created. This is not what we want, as at least one approval needs to come from a _maintainer_.

https://github.com/ceph/ceph-csi/blob/1dc032e5545cd5296b272e755aeda5d6bf4c9af5/.mergify.yml#L35-L57

The above rule (and duplicates) needs to get adjusted to:

```yaml
  - name: automatic merge
    conditions:
      - label!=DNM
      - base~=^(devel)|(release-.+)$
      - approved-reviews-by=@core-contributors
      - approved-reviews-by=@maintainers
      - "#changes-requested-reviews-by=0"
```

- `approved-reviews-by` is a list as described in the [attributes](https://docs.mergify.io/conditions/#attributes)
- contents of lists are [matched with a simple `=`](https://docs.mergify.io/conditions/#how-to-match-lists)

### PR DESCRIPTION
**Pull request by Madhu-1**
updated mergify rules to consider the team's approval to merge a PR.

more details at #2367

fixes #2367

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
CC @ceph/ceph-csi-contributors @ceph/ceph-csi-maintainers

File: .mergify.yml
Comment (nixpanic) [inline .mergify.yml:41]: I wonder if the notation should be `@ceph-csi-contributors` without the `ceph/` organization part. Did you find an example configuration file where teams are used?

Comment (Madhu-1) [reply .mergify.yml:41]: took example from https://github.com/Mergifyio/mergify-engine/issues/310#issuecomment-444419127

Comment (Rakshith-R) [reply .mergify.yml:41]: I guess @ceph/ceph-csi-contributors  is working 
https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2425

**Comment by Madhu-1** [comment]
This needs a manual merge. @humblec @nixpanic feel free to merge this PR after approval.

**Comment by Rakshith-R** [comment]
@Madhu-1 , we need to preserve `- "#approved-reviews-by>=2"`
since a maintainer is also a contributor, its being counted twice, details in the link

https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2423

**Comment by Madhu-1** [comment]
> @Madhu-1 , we need to preserve `- "#approved-reviews-by>=2"`
> since a maintainer is also a contributor, its being counted twice, details in the link
> 
> https://dashboard.mergify.io/github/ceph/config-editor?repositoryName=ceph-csi&pullRequestNumber=2423

Yes correct. updating it

File: .mergify.yml
Comment (Rakshith-R) [inline .mergify.yml:129]:
```suggestion
      - "#approved-reviews-by>=2"
      - "approved-reviews-by=@ceph/ceph-csi-contributors"
      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
```

Backport prs also need two approvals now ?

Comment (Madhu-1) [reply .mergify.yml:129]: Yes as we have more people let's have 2 approvals

**Comment by humblec** [comment]
@Madhu-1 i think its good to update the contribution guide with this too?

**Review by Rakshith-R — COMMENTED** [review]
Same changes missing for backport release v1.2 branch (line 105/107)but maybe remove backport pr rules for v1.2 completely too? 
Otherwise LGTM.

**Comment by Madhu-1** [comment]
> @Madhu-1 i think its good to update the contribution guide with this too?

updated @humblec PTAL

**Comment by Madhu-1** [comment]
> Same changes missing for backport release v1.2 branch (line 105/107)but maybe remove backport pr rules for v1.2 completely too?
> Otherwise LGTM.

v1.x is getting deprecated will remove all mergify rules for that in as a follow-up PR.

**Review by Rakshith-R — APPROVED** [review]

**Comment by humblec** [comment]
@nixpanic are you good with the changes here?

**Review by nixpanic — APPROVED** [review]

**Comment by Madhu-1** [comment]
Merging the PR as it has required approvals.

**Comment by Madhu-1** [comment]
> Sorry but I didn't understand the command. Please consult [the commands documentation](https://docs.mergify.io/commands.html) .
> 
> _Hey, I reacted but my real name is @Mergifyio_

issue already reported for this at https://github.com/Mergifyio/mergify-engine/issues/3068

**Pull request merged**
